corehq/messaging/scheduling/views.py

```diff
@@ -890,7 +890,16 @@ def post(self, request, *args, **kwargs):
                 self.criteria_form.save_criteria(rule)
                 self.schedule_form.save_rule_action_and_schedule(rule)
 
-            initiate_messaging_rule_run(rule)
+            # list of fields that are getting changed everytime a request came manually from django form.
+            changed_schedule_fields = [
```

Comment:
Out of curiosity, where did this list of fields come from?

Reply:
Am I understanding correctly that these are all fields that always change, even if the user didn't take any action? And they're also fields where it's impossible or the user to change them without changing some other field that will be detected?

Spot checking, that's what most of them look like, though `stop_type` is one I'm not sure of - couldn't you change `stop_type` from after to never without changing anything else in the form?

Reply:
yes, you are right, I am getting list of all the changed field using method Form.change_data without doing any configuration changes from UI side and stored it into a changed_schedule_field list. next time when request comes I am only considering fields which does not belongs to this list. stop_type also changes so it's not reliable.

Reply:
I believe making this form to not save in case its not needed, is going to need a really thorough exercise and should not be done here unless one has really understood well how these forms are plugged together. So for now keeping just the modal sounds good.

There was a PR to not trigger in case only translations are changed, which we were doing via JS. That was never concluded right?

Reply:
@mkangia are you worried that this code is buggy, or will be hard to maintain, or something else?

Reply:
@orangejenny Both.
Definitely hard to maintain, each time a new field is added in the forms, someone would definitely forget to update it here. Though making this happen in the form definition would make it better but might be difficult to setup considering the number of forms here.

Are we saying that these fields always change, so if the user has edited anything apart from these, only these initiate the rule?
If yes, that does not sound right. Should not this list be the list of fields that don't need re-processing, and hence then only reprocess if there are more fields apart from those. My thought is that getting that list at the first place from the forms should be flexible and robust.

Overall, I don't think the users know or have enough knowledge to know when and if an alert needs to reprocess, so if there is a miss here, it will just silently go wrong (process or not process). And hence this needs to be really robust.

Reply:
@mkangia Thank you for raising. Just pinged you on the ticket about this - I think your concerns make sense.